llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -2985,6 +2985,14 @@ Instruction *InstCombinerImpl::foldSelectOfBools(SelectInst &SI) {
       return BinaryOperator::CreateOr(CondVal, FalseVal);
     }
 
+    if (match(CondVal, m_LogicalOr(m_Value(A), m_Value(B))) &&
+        isa<SelectInst>(CondVal) && impliesPoison(FalseVal, B)) {
```

Comment:
```suggestion
    if (match(CondVal, m_Select(m_Value(A), m_One(), m_Value(B))) && impliesPoison(FalseVal, B)) {
```
Please add some negative tests:
+ multi-use
+ bitwise or